### DISCUSSION TITLE
Obsidian Modal Forms Plugin: How to Transform your Data Entry in Obsidian

### DISCUSSION THREAD

**Discussion by motion2082** [danielo515/obsidian-modal-form#354] [Show and tell]
![Thumbnail02_compressed](https://github.com/user-attachments/assets/1735e72b-a0fd-4819-bbcd-8a8fdbd7cdce)
## Description
In today’s video, we're diving deep into the Modal Form Plugin - a powerful tool that may transform how you capture and organise information in your Obsidian Vault. 

This video is perfect for Obsidian users who are familiar with the QuickAdd and Templater plugins. We'll explore how to leverage advanced Templater templates alongside Modal Forms to supercharge your note-taking workflow.

Looking for a seamless way to input data into your Obsidian Vault? Join me as we unlock the full potential of the Modal Forms Plugin

## Shout Out
Thank you Danielo for all your hard work

Regards,
Paul

**Comment by danielo515** [comment]
What a wonderful and comprehensive guide! Better than I could have ever done, thank you!
I see you read the docs completely! I am amazed.
I have to admit that at some points I was thinking:

> He is surely missing that you can trigger forms from captures

But then you mention it, you were just going step by step exploring all the options, I take my hat off 🎩
I love to see how people is using the plugin and how they perceive the features. Also, that gives me a hint of where things can be improved or be made simpler.
Also, it bring me to tears when I see you using both the simple templates and the templater editor. I thought nobody was using those, I'm so happy 😭 !

By the way, watching your video a little tip came to my mind.
When you already have a long list of values that you want in a select input, instead of putting them one by one in the fixed list, you can use the `dataview` input type and just return the values as an array in the dataview query.

For example, you can put the following in a select input with `dataview` as the source:

```ts
return [
    'Wash the dishes',
    'Take out the trash',
    'Do the laundry',
    'Mow the lawn',
    'Feed the cat',
    'Feed the dog',
    'Feed the fish',
    'Feed the birds',
    'Feed the turtles',
    'Feed the rabbits',
    'Feed the turtles',]
```

And the result will be:
<img width="618" alt="image" src="https://github.com/user-attachments/assets/a8cb18b9-6c88-4a31-a069-28a602d8d7ca" />


This is how it looks in the form builder:
<img width="703" alt="image" src="https://github.com/user-attachments/assets/0692256f-75e9-4b3d-9ebd-f06243042b39" />

Again, thank you very much for your great guide

**Comment by motion2082** [comment]
Hi Danielo,

> What a wonderful and comprehensive guide! Better than I could have ever done, thank you!
> I see you read the docs completely! I am amazed.

Thank you. 
To fully understand the plugin I spent a lot of time reading the documentation which is really helpful. I don't think it would have been possible if you hadn't put in the effort to document everything.  

> I have to admit that at some points I was thinking:
> He is surely missing that you can trigger forms from captures
> But then you mention it, you were just going step by step exploring all the options, I take my hat off 

This was by design to showcase the different methods and way to integrate Modal Forms :)

> I love to see how people is using the plugin and how they perceive the features. Also, that gives me a hint of where things can be improved or be made simpler.

Initially it was a bit of a learning curve and I was struggling a little to get my head around what you had created. I believe a conversation in another thread was the lightbulb moment where I was like, now I understand, this is worth sharing. 

> Also, it bring me to tears when I see you using both the simple templates and the templater editor. I thought nobody was using those, I'm so happy 😭 !

That's really great to hear. Nothing like seeing your hard work making a difference to peoples lives. The template editor I find really useful now I understand it but initially the simple templater seemed confusing and not easily integrated with existing templater syntax.

> By the way, watching your video a little tip came to my mind.

Thank you for this suggestion, I had a feeling there would have been a better way to do it. Maybe can share that tip in a future video :)

> Again, thank you very much for your great guide

Your welcome, hopefully it helps more people discover your plugin and provides some additional data entry options that QuickAdd and Templater can't provide.

**Reply by danielo515** [reply]
> Initially it was a bit of a learning curve and I was struggling a little to get my head around what you had created

Yes, that is the main problem of this plugin, the entry point.
I am good at identifying problems and finding solutions for them, but I'm not bery good at marketing those solutions to people with similar problems.
I'm thinking about creating a better headline that better reflects the intention. Maybe something like "Complex and visually pleasing data collection for templater" or something like that.

Again, thanks for your feedback
